### ISSUE TITLE
Obfuscation Failing

### ISSUE THREAD

**Issue by Sequence5424** [sabnzbd/sabnzbd#2541]
Hello!
I've noticed sometimes sabnzbd fails to deobfuscate and enabled logging to `+DEBUG` lately and noticed something weird in the log and that is
```
DEBUG::[deobfuscate_filenames:111] Checking: <HASH>ed1fc8e.vol000+001
DEBUG::[deobfuscate_filenames:158] Not obfuscated: (upperchars + lowerchars >= 4) and decimals > 3 and spacesdots > 1
```
So, it detects that the file name isn't obfuscated, but the file name was `0e72af8afed74cea9ef7beb69ed1fc8e.mkv` in the end.

Would appreciate any help.

Thank you!

**Comment by sanderjo** [comment]
That first line is just an intermediate file. Probably a par2. So that doesn't matter.

The final file `0e72af8afed74cea9ef7beb69ed1fc8e.mkv` is certainly obfuscated, so SAB should deobfuscate that. So can you find & post the relevant logging?

**Comment by Sequence5424** [comment]
> Unfortunately, I think Sorting is the problem. We never implemented Deobfuscate for Sorting, because if Sorting is active it should handle the file renaming and not Deobfuscate. Is it possible to share a (pastebin) link to the whole log of the whole post processing?

Sure, here is a link, I hope I didn't leave out sensitive information.
https://paste.sh/-p3llkaE#4D8pmxEzUzpUQACyG1uGsHKE

**Comment by Safihre** [comment]
And what are your Movie Sorting settings?

**Comment by Sequence5424** [comment]
> And what are your Movie Sorting settings?

`%title (%y)/%fn.%ext`
![image](https://user-images.githubusercontent.com/128895401/234104775-5eae0f81-0c0a-4ab5-8648-1e07887ed492.png)

**Comment by Safihre** [comment]
Right. Why %fn? That preserves the original filename, which in your case is obfuscated.
You should use the %title one or %sn.

**Comment by thezoggy** [comment]
you prob want the job name `%dn` - but if your using 3rd party apps like radarr/sonarr. you should NOT be using any sorting with sab and instead just let those apps handle it on their side.

**Comment by Sequence5424** [comment]
I'm not sure, but one of the reasons I'm not using directly `%dn` is because there are some files that the nzb file changes name to different format, and I prefer to have a format which is "more universal".
The confusing part is that `%fn` is working fine sometimes, and sometimes it isn't, Like I can tell if its deobfuscating because If I try viewing the file structure in sabnzbd, I can see random letters but, in the end, if I check the file name when it's done, the file name is fixed.
And I'm not using any 3rd party app, which is why I'm really dependent on the sorting feature of sabnzbd

**Comment by Safihre** [comment]
Alright, you will just have to either use `%dn` to keep the original name of the download and otherwise `%title`.
It just doesn't make sense to Deobfuscate with Sorting turned on.
I will add a note to the wiki about this.

**Comment by thezoggy** [comment]
could the user just take the old deobfuscate script and set that as the post processing script for that cat and get it to just run regardless?
